internal/provider/resource_edge_cluster.go

```diff
@@ -0,0 +1,250 @@
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name for the Tier-0 gateway",
+			},
+			"tier1_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name for the Tier-1 gateway",
+			},
+			"profile_type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "One among: DEFAULT, CUSTOM. If set to CUSTOM a 'profile' must be provided",
+			},
+			"profile": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				MaxItems:    1,
+				Description: "The specification for the edge cluster profile",
+				Elem:        edge_cluster.ClusterProfileSchema(),
+			},
+			"routing_type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "One among: EBGP, STATIC",
+			},
+			"form_factor": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "One among: XLARGE, LARGE, MEDIUM, SMALL",
+			},
+			"high_availability": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "One among: ACTIVE_ACTIVE, ACTIVE_STANDBY",
+			},
+			"mtu": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: "Maximum transmission unit size for the cluster",
+			},
+			"asn": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "ASN for the cluster",
+			},
+			"skip_tep_routability_check": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Set to true to bypass normal ICMP-based check of Edge TEP / host TEP routability (default is false, meaning do check)",
+			},
+			"tier1_unhosted": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Select whether Tier-1 being created per this spec is hosted on the new Edge cluster or not (default value is false, meaning hosted)",
+			},
+			"internal_transit_subnets": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Subnet addresses in CIDR notation that are used to assign addresses to logical links connecting service routers and distributed routers",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"transit_subnets": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Transit subnet addresses in CIDR notation that are used to assign addresses to logical links connecting Tier-0 and Tier-1s",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"edge_node": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "The nodes in the edge cluster",
+				Elem:        edge_cluster.EdgeNodeSchema(),
+			},
+		},
+	}
+}
+
+func resourceEdgeClusterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*api_client.SddcManagerClient).ApiClient
+
+	params := &nsxt_edge_clusters.CreateEdgeParams{
+		EdgeCreationSpec: edge_cluster.GetEdgeClusterCreationSpec(d),
+		Context:          ctx,
+	}
```

Comment:
add WithTimeout(constants.DefaultVcfApiCallTimeout)

Reply:
done, added it to all API calls